cli/src/semgrep/meta.py

```diff
@@ -244,12 +244,32 @@ def _get_latest_commit_hash_in_branch(self, branch_name: str) -> str:
     def head_branch_hash(self) -> str:
         """
         Commit of the head branch, reported via the GitHub pull_request event
+        This will also ensure that a fetch is done prior to returning.
 
         Assumes we are in PR context
         """
         head_branch_name = self._head_branch_ref
         commit = self.glom_event(T["pull_request"]["head"]["sha"])
-        logger.debug(f"head branch ({head_branch_name}) has latest commit {commit}")
+        logger.debug(
+            f"head branch ({head_branch_name}) has latest commit {commit}, fetching that commit now."
+        )
+        process = subprocess.run(
```

Comment:
We have a helper function, `git_check_output`, which prints a nice error message if git is not available; we should use that here.

Reply:
changed